### ISSUE TITLE
Better way to save a model in the scikit-learn tuner

### ISSUE THREAD

**Issue by qingquansong** [keras-team/keras-tuner#462]
The current scikit-learn tuner use pickle to save the model after running every trial, but pickle does not allow local object to be saved such as a local function or class. Can we use [dill](https://dill.readthedocs.io/en/latest/dill.html) or other way to save the model? One of the benefits of this would be the available of joint model selections among keras models and shallow models in scikit-learn. An example code is listed below:

from tensorflow import keras
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
from kerastuner.engine import hyperparameters as hp
from tensorflow.keras.wrappers.scikit_learn import KerasClassifier  

def build_model(hp):
    model_type = hp.Choice('model_type', ['svm', 'random_forest', "mlp"])
    if model_type == 'svm':
        with hp.conditional_scope('model_type', 'svm'):
            model = SVC(C=hp.Choice("C", [0.05, 0.1, 0.2, 0.5, 1.0, 2.0, 5.0], default=1),
                        kernel=hp.Choice("kernel_type", ['linear', 'rbf'], default="linear"), 
                        random_state=42
                       )
    elif model_type == 'random_forest':
        with hp.conditional_scope('model_type', 'random_forest'):
            model = RandomForestClassifier(
                n_estimators=hp.Int('n_estimators', 10, 50, step=10),
                max_depth=hp.Int('max_depth', 3, 10))
    elif model_type == 'mlp':
        with hp.conditional_scope('model_type', 'mlp'):
          units = hp.Choice("units", [8, 16, 32], default=8)
          def build_mlp1():
              model = keras.Sequential([
                  keras.layers.Flatten(input_shape=X_train.shape[1:]),
                  keras.layers.Dense(units, activation='relu'),
                  keras.layers.Dense(10, activation='softmax')
              ])
              model.compile(optimizer='adam',
                        loss=keras.losses.SparseCategoricalCrossentropy(),
                        metrics=['accuracy'])
              return model
          model = KerasClassifier(build_mlp1, epochs=5, batch_size=64)
    else:
        raise ValueError('Unrecognized model_type')
    return model

tuner = kt.tuners.Sklearn(
        oracle=kt.oracles.BayesianOptimization(
            objective=kt.Objective('score', 'max'),
            max_trials=10),
        hypermodel=build_model)
tuner.search(X_train, y_train)
